src/tools/pkcs11-tool.c

```diff
@@ -4442,6 +4442,18 @@ static void gost_info_free(struct gostkey_info gost)
 	OPENSSL_free(gost.public.value);
 	OPENSSL_free(gost.private.value);
 }
+
+static void rsa_info_free(struct rsakey_info rsa)
+{
+	OPENSSL_free(rsa.modulus);
```

Comment:
the memory of all members is allcated using `malloc()` (see line 4035); please use `free()` for releasing it.

Reply:
If that's the case, free() should also be used in gost_info_free(), as they are also allocated with malloc(), see line 4147, 4163, 4197.